webvh/webvh/did/manager.py

```diff
@@ -0,0 +1,635 @@
+                ssl=False,
+            )
+            response_json = await response.json()
+            if response.status == http.HTTPStatus.BAD_REQUEST:
+                raise DidCreationError(response_json.get("detail"))
+
+            log_entry = response_json.get("logEntry")
+
+            if not author_key_info:
+                author_key_info = await self._get_or_create_author_key(
+                    endorsed_document["id"]
+                )
+
+            # Add author proof to the log entry
+            signed_log_entry = await DataIntegrityManager(session).add_proof(
+                log_entry,
+                DataIntegrityProofOptions(
+                    type="DataIntegrityProof",
+                    cryptosuite="eddsa-jcs-2022",
+                    proof_purpose="assertionMethod",
+                    verification_method=f"did:key:{author_key_info.get('multikey')}#{author_key_info.get('multikey')}",
+                ),
+            )
+
+            id_parts = endorsed_document["id"].split(":")
+            namespace = id_parts[-2]
+            identifier = id_parts[-1]
+
+            # Submit the initial log entry
+            response = await http_session.post(
+                f"{server_url}/{namespace}/{identifier}",
+                json={"logEntry": signed_log_entry},
+                ssl=False,
+            )
+            response_json = await response.json()
+            if response.status == http.HTTPStatus.BAD_REQUEST:
+                raise DidCreationError(response_json.get("detail"))
+            did = response_json.get("state", {}).get("id")
+
+            # Save the did in the wallet
+            await session.handle.insert(
+                CATEGORY_DID,
+                did,
+                value_json={
+                    "did": did,
+                    "verkey": multikey_to_verkey(author_key_info.get("multikey")),
+                    "metadata": {
+                        "posted": True,
+                        "namespace": namespace,
+                        "identifier": identifier,
+                    },
+                    "method": "webvh",
+                    "key_type": "ed25519",
+                },
+                tags={},
+            )
+            resolver = session.inject(DIDResolver)
+
+            resolved_did_doc = (
+                await resolver.resolve_with_metadata(self.profile, did)
+            ).serialize()
+
+            event_bus = self.profile.inject(EventBus)
+
+            metadata = resolved_did_doc["metadata"]
+            metadata["state"] = "posted"
+            await event_bus.notify(
+                self.profile,
+                Event(
+                    f"endorsement_response::{endorsed_document['id']}",
+                    {"document": resolved_did_doc["did_document"], "metadata": metadata},
+                ),
+            )
+
+        return resolved_did_doc
+
+    async def update(self, options: dict, features: dict):
```

Comment:
currently not implemented on the didwebvh server. The route exists, but will return a 501

Reply:
Ok. The way this was working was by creating the next custom log entry and posting it to the /namespace/identifier endpoint.

Do we need another endpoint for this?

Reply:
updates will be using the `PUT` method